tests/Unit/Crypt/HashTest.php

```diff
@@ -0,0 +1,220 @@
+                '773ea91e36800e46854db8ebd09181a72959098b3ef8c122d9635514ced565fe',
+            ),
+            // Test Case 4
+            array(
+                'sha256',
+                pack('H*', '0102030405060708090a0b0c0d0e0f10111213141516171819'),
+                pack('H*', 'cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd'),
+                '82558a389a443c0ea4cc819899f2083a85f0faa3e578f8077a2e3ff46729665b',
+            ),
+            // RFC 4231
+            // Test Case 1
+            array(
+                'sha512',
+                pack('H*', '0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b'),
+                pack('H*', '4869205468657265'),
+                '87aa7cdea5ef619d4ff0b4241a1d6cb02379f4e2ce4ec2787ad0b30545e17cdedaa833b7d6b8a702038b274eaea3f4e4be9d914eeb61f1702e696c203a126854',
+            ),
+            // Test Case 2
+            array(
+                'sha512',
+                pack('H*', '4a656665'),
+                pack('H*', '7768617420646f2079612077616e7420666f72206e6f7468696e673f'),
+                '164b7a7bfcf819e2e395fbe73b56e0a387bd64222e831fd610270cd7ea2505549758bf75c05a994a6d034f65f8f0e6fdcaeab1a34d4a6b4b636e070a38bce737',
+            ),
+            // Test Case 3
+            array(
+                'sha512',
+                pack('H*', 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'),
+                pack('H*', 'dddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd'),
+                'fa73b0089d56a284efb0f0756c890be9b1b5dbdd8ee81a3655f83e33b2279d39bf3e848279a722c806b485a47e67c807b946a337bee8942674278859e13292fb',
+            ),
+            // Test Case 4
+            array(
+                'sha512',
+                pack('H*', '0102030405060708090a0b0c0d0e0f10111213141516171819'),
+                pack('H*', 'cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd'),
+                'b0ba465637458c6990e5a8c5f61d4af7e576d97ff94b872de76f8050361ee3dba91ca5c11aa25eb4d679275cc5788063a5f19741120c4f2de2adebeb10a298dd',
+            ),
+            array(
+                'whirlpool',
+                'abcd',
+                'The quick brown fox jumps over the lazy dog',
+                'e71aabb2588d789292fa6fef00b35cc269ec3ea912b1c1cd7127daf95f004a5df5392ee563d322bac7e19d9eab161932fe9c257d63e0d09eca0d91ab4010125e',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider hashData()
+     */
+    public function testHash($hash, $message, $result)
+    {
+        $this->assertHashesTo($hash, $message, $result);
+    }
+
+    /**
+     * @dataProvider hashData()
+     */
+    public function testHash96($hash, $message, $result)
+    {
+        $this->assertHashesTo($hash . '-96', $message, substr($result, 0, 24));
+    }
+
+    public function testGetHash()
+    {
+        $hash = new Hash();
```

Comment:
If you're bored, please try make individual test methods and describe what you're testing. E.g.
```
public function testConstructorDefault()
{
    $hash = new Hash();
    $this->assertEquals($hash->getHash(), 'sha256');
}
```